code/modules/surgery/surgery_step.dm

```diff
@@ -69,19 +70,28 @@
 	if(tool)
 		speed_mod = tool.toolspeed
 
-	if(do_after(user, time * speed_mod, target = target))
-		var/prob_chance = 100
+	var/implement_speed_mod = 1
+	if(implement_type)	//this means it isn't a require hand or any item step.
+		implement_speed_mod = implements[implement_type] / 100.0
 
-		if(implement_type)	//this means it isn't a require hand or any item step.
-			prob_chance = implements[implement_type]
-		prob_chance *= surgery.get_propability_multiplier()
+	speed_mod /= (get_location_modifier(target) * (1 + surgery.speed_modifier) * implement_speed_mod)
+	var/modded_time = time * speed_mod
+
+	fail_prob = min(max(0, modded_time - (time * 2)),99)//if modded_time > time * 2, then fail_prob = modded_time - time*2. starts at 0, caps at 99
+	modded_time = min(modded_time, time * 2)//also if that, then cap modded_time at time*2

```

Comment:
To any future people who want to tweak these numbers, the * 2 on this line and the line above can be increased to make surgery go slower but fail less, and decreased to make surgery faster but fail more. I wouldn't recommend lowering it and anything above 3x seems excessive to me.

Reply:
should define 😏

Reply:
>wasting a whole milisecond of compile time and a whole line of code to define a value that's used twice within a line from itself

😬

Reply:
beauty comes at a price